static/js/public/manage-snaps.js

```diff
@@ -0,0 +1,58 @@
+function getSelectedSnaps(snapCheckboxes) {
+  let snaps = [];
+
+  snapCheckboxes.forEach((snap) => {
+    if (snap.checked) {
+      snaps.push({
+        name: snap.dataset.snapName,
+      });
+    }
+  });
+
+  return snaps;
+}
+
+function updateSnaps() {
+  const manageSnapsForm = document.querySelector("#manage-snaps-form");
+  const saveButton = document.querySelector(".js-save-button");
+  const snapCheckboxes = Array.prototype.slice.call(
+    document.querySelectorAll(".js-snap-checkbox")
+  );
+
+  const originalSnapState = getSelectedSnaps(snapCheckboxes);
+  let currentSnaps = getSelectedSnaps(snapCheckboxes);
```

Comment:
You could save the compute here by just assigning the original variable above

Reply:
Good point. Done.